airbyte-oauth/src/main/java/io/airbyte/oauth/OAuthImplementationFactory.java

```diff
@@ -8,63 +8,65 @@
-import io.airbyte.oauth.flows.TrelloOAuthFlow;
+import io.airbyte.oauth.flows.*;
 import io.airbyte.oauth.flows.facebook.FacebookMarketingOAuthFlow;
 import io.airbyte.oauth.flows.facebook.FacebookPagesOAuthFlow;
 import io.airbyte.oauth.flows.facebook.InstagramOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleAdsOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleAnalyticsOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleSearchConsoleOAuthFlow;
 import io.airbyte.oauth.flows.google.GoogleSheetsOAuthFlow;
+import java.lang.reflect.InvocationTargetException;
 import java.net.http.HttpClient;
 import java.util.Map;
-import java.util.UUID;
 
-public class OAuthImplementationFactory {
+public class OAuthImplementationFactory<T extends OAuthFlowImplementation> {
 
-  private final Map<String, OAuthFlowImplementation> OAUTH_FLOW_MAPPING;
+  private final Map<String, Class<? extends OAuthFlowImplementation>> OAUTH_FLOW_MAPPING;
+  private final ConfigRepository configRepository;
+  private final HttpClient httpClient;
 
   public OAuthImplementationFactory(final ConfigRepository configRepository, final HttpClient httpClient) {
-    OAUTH_FLOW_MAPPING = ImmutableMap.<String, OAuthFlowImplementation>builder()
-        .put("airbyte/source-asana", new AsanaOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-facebook-marketing", new FacebookMarketingOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-facebook-pages", new FacebookPagesOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-github", new GithubOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-google-ads", new GoogleAdsOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-google-analytics-v4", new GoogleAnalyticsOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-google-search-console", new GoogleSearchConsoleOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-google-sheets", new GoogleSheetsOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-hubspot", new HubspotOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-intercom", new IntercomOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-instagram", new InstagramOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-salesforce", new SalesforceOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-slack", new SlackOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-surveymonkey", new SurveymonkeyOAuthFlow(configRepository, httpClient))
-        .put("airbyte/source-trello", new TrelloOAuthFlow(configRepository, httpClient))
+    this.configRepository = configRepository;
+    this.httpClient = httpClient;
+    OAUTH_FLOW_MAPPING = ImmutableMap.<String, Class<? extends OAuthFlowImplementation>>builder()
+        .put("airbyte/source-asana", AsanaOAuthFlow.class)
```

Comment:
so you want to create new instance on every call? It  It would break trello cause it using oAuth1.0 and storing some parameters from previos step.

Reply:
Looks like I was wrong, it wont break trello cause it already using static variables for saving state. Can we make OAUTH_FLOW_MAPPING static final ?